api/src/contracttest/java/org/pmiops/workbench/consumer/RuntimesApiTest.java

```diff
@@ -104,78 +135,147 @@ RequestResponsePact getMissingRuntime(PactDslWithProvider builder) {
-                            context.stringType("creator", "Bugs Bunny");
-                            context.stringType("createdDate", "Yesterday");
-                            context.stringType("dateAccessed", "Tuesday");
-                          });
+                          "labelsToUpsert", labels -> labels.stringValue("key1", "ke1Updated"));
                     })
                 .build())
+        .willRespondWith()
+        .status(404)
+        .toPact();
+  }
+
+  @Pact(consumer = "aou-rwb-api", provider = "leonardo")
+  RequestResponsePact deleteRuntime(PactDslWithProvider builder) {
+    return builder
+        .given("there is a runtime in a Google project")
+        .uponReceiving("a request to delete a runtime")
+        .method("DELETE")
+        .path("/api/google/v1/runtimes/googleProject/runtimename")
+        .query("deleteDisk=true")
+        .willRespondWith()
+        .status(202)
+        .toPact();
+  }
+
+  @Pact(consumer = "aou-rwb-api", provider = "leonardo")
+  RequestResponsePact deleteMissingRuntime(PactDslWithProvider builder) {
+    return builder
+        .given("there is not a runtime in a Google project")
+        .uponReceiving("a request to delete a runtime")
+        .method("DELETE")
+        .path("/api/google/v1/runtimes/googleProject/runtimename")
+        .query("deleteDisk=true")
+        .willRespondWith()
+        .status(404)
+        .toPact();
+  }
+
+  @Pact(consumer = "aou-rwb-api", provider = "leonardo")
+  RequestResponsePact stopRuntime(PactDslWithProvider builder) {
+    return builder
+        .given("there is a runtime in a Google project")
+        .uponReceiving("a request to stop a runtime")
+        .method("POST")
+        .path("/api/google/v1/runtimes/googleProject/runtimename/stop")
+        .willRespondWith()
+        .status(202)
         .toPact();
   }
 
-  //  @Pact(consumer = "aou-rwb-api", provider = "leonardo")
-  //  RequestResponsePact updateMissingRuntime(PactDslWithProvider builder) {
-  //    return builder
-  //        .given("there is not a runtime in a Google project")
-  //        .uponReceiving("a request to get that runtime from GSuite")
-  //        .method("PATCH")
-  //        .path("/api/google/v1/runtimes/googleProject/runtimeName")
-  //        .willRespondWith()
-  //        .status(404)
-  //        .headers(contentTypeJsonHeader)
-  //        .body(newJsonBody(body -> {}).build())
-  //        .toPact();
-  //  }
+  //
+  @Pact(consumer = "aou-rwb-api", provider = "leonardo")
+  RequestResponsePact stopMissingRuntime(PactDslWithProvider builder) {
+    return builder
+        .given("there is not a runtime in a Google project")
+        .uponReceiving("a request to stop a runtime")
+        .method("POST")
+        .path("/api/google/v1/runtimes/googleProject/runtimename/stop")
+        .willRespondWith()
+        .status(404)
+        .toPact();
+  }
 
   @Test
   @PactTestFor(pactMethod = "createNewRuntime")
```

Comment:
code style idea: this file is organized as 10 pacts and then the corresponding 10 pact tests (apologies if the terminology is wrong) - what do you think about alternating pact / test / pact / test, so the context is kept nearby?

Reply:
I like that! Thanks for the suggestion.